### PR TITLE
Add `:executable` option

### LINKED ISSUE #3: Add `:executable` config option
We use [Mint](http://github.com/yonaskolb/Mint) for installing our Swift command line dependencies and as a result, would need to invoke `mint run yonaskolb/XcodeGen xcodegen ...` rather than just `xcodegen` when performing the `sh` command since its not installed globally... 

For this plugin to support this use case, we should add the capability to to specify the executable that is used in a similar way to the [`swiftlint`](https://docs.fastlane.tools/actions/swiftlint/) action.

(Note: I'm happy picking this up, I just wanted to raise tickets to the improvements that I'm looking to make ahead of implementing them to get feedback if required 🙇)

### PR DESCRIPTION
**Pull request by liamnichols**
Closes #3 

**Note: This change depends on #5, I'll rebase this PR after that PR merges. You can view the actual diff [here](https://github.com/NovatecConsulting/fastlane-plugin-xcodegen/pull/6/files/4327fd8f4489187897016ffdda18fb926abafa3c..c83e936d003cb8a0dac4cef3a6588ea3ee315655) in the meantime.**

In this change, I do a couple of things:

1. Add a test to ensure that we install XcodeGen when `which xcodegen` fails
2. Add the `:executable` option, don't attempt to install/check for XcodeGen when set
3. Use the `:executable` value instead of `"xcodegen"` for the command invoked when calling `sh` if set 
4. Add a test to ensure the new behaviour

### CODE REVIEW

File: lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb [hunk 1]
```diff
@@ -42,6 +42,10 @@ def self.details
 
       def self.available_options
         [
+          FastlaneCore::ConfigItem.new(key: :executable,
```
Comment (m-ruhl) [inline lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb:45]: Can you add the new option to the README file

**Review by m-ruhl — COMMENTED** [review]
Hi @liamnichols,

thanks for the PR.. was not aware of mint :) 

Besides the README-file it looks good to me.

File: lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb (hunk 1 above)
Comment (liamnichols) [reply lib/fastlane/plugin/xcodegen/actions/xcodegen_action.rb:45]: Sure!

**Comment by liamnichols** [comment]
Thanks @m-ruhl, i've updated the README and rebased from `master` 🙏

**Pull request merged**